OpenRA.Mods.Common/Traits/BotModules/BotModuleLogic/BaseBuilderQueueManager.cs

```diff
@@ -364,15 +389,56 @@ ActorInfo ChooseBuildingToBuild(ProductionQueue queue)
 			}
 
 			// Too spammy to keep enabled all the time, but very useful when debugging specific issues.
-			// AIUtils.BotDebug("{0} couldn't decide what to build for queue {1}.", queue.Actor.Owner, queue.Info.Group);
+			// AIUtils.BotDebug($"{queue.Actor.Owner} couldn't decide what to build for queue {queue.Info.Group}.");
 			return null;
 		}
 
+		CPos? ChooseBarrierLocation(ActorInfo wallInfo)
+		{
+			CPos? FindUnwalledEdgeAround(Actor actor)
+			{
+				if (actor == null || actor.Info == null)
+					return null;
+
+				var buildingInfo = actor.Info.TraitInfoOrDefault<BuildingInfo>();
+				if (buildingInfo == null)
+					return null;
+
+				var topLeft = actor.Location;
+				var edges = new List<CPos>()
+				{
+					topLeft + new CVec(-1, -1),
+					topLeft + new CVec(buildingInfo.Dimensions.X, -1),
+					topLeft + new CVec(-1, buildingInfo.Dimensions.Y),
+					topLeft + buildingInfo.Dimensions,
+				};
+
+				if (buildingInfo == null)
+					return null;
+
+				var wallBuildingInfo = wallInfo.TraitInfoOrDefault<BuildingInfo>();
+				if (wallBuildingInfo == null)
+					return null;
+
+				var possibleEdges = edges.Where(e => world.CanPlaceBuilding(e, wallInfo, wallBuildingInfo, null));
+				if (!possibleEdges.Any())
+					return null;
+
+				return possibleEdges.Random(world.LocalRandom);
+			}
+
+			// Build around important structure
+			var buildingToWall = world.ActorsHavingTrait<Building>().FirstOrDefault(a => !a.Disposed && a.Owner == player &&
+				baseBuilder.Info.WalledStructures.Contains(a.Info.Name));
+
+			return FindUnwalledEdgeAround(buildingToWall);
```

Comment:
The method is only used once, so why not inline it here? (That also resolves the comment above, since the method cannot be called with anything else than a building if it is inlined here.)